### PR TITLE
HIVE-27441: Iceberg: CTLT with source table as Hive managed table fails with "The table must be stored using an ACID compliant format"

### PR DESCRIPTION
**Pull request by ayushtkn**
### What changes were proposed in this pull request?

Remove any transactional property from the iceberg table in case of CTLT, as they aren't relevant to it.

### Why are the changes needed?

CTLT with iceberg table as target and source as managed table fails, due to additional checks that happen due to transactional properties set in the iceberg tables

### Does this PR introduce _any_ user-facing change?

Yes, CTLT from managed to iceberg table will be successful 

### Is the change a dependency upgrade?

No

### How was this patch tested?

UT

### CODE REVIEW

**Review by aturoczy — APPROVED** [review]
+1

File: iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java [hunk 1]
```diff
@@ -1381,6 +1381,11 @@ public void setTableParametersForCTLT(org.apache.hadoop.hive.ql.metadata.Table t
       SessionStateUtil.addResourceOrThrow(conf, hive_metastoreConstants.PARTITION_TRANSFORM_SPEC, spec);
       tbl.getSd().getCols().addAll(tbl.getPartitionKeys());
       tbl.getTTable().setPartitionKeysIsSet(false);
+
+      // Remove any transactional properties as they aren't relevant in case of iceberg tables.
+      tbl.getParameters().keySet().removeIf(
+          key -> hive_metastoreConstants.TABLE_IS_TRANSACTIONAL.equalsIgnoreCase(key) ||
+              hive_metastoreConstants.TABLE_TRANSACTIONAL_PROPERTIES.equalsIgnoreCase(key));
```
Comment (deniskuzZ) [inline iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java:1388]: should we filter out `hive_metastoreConstants.TABLE_TRANSACTIONAL_PROPERTIES` as well?

**Review by simhadri-g — APPROVED** [review]
LGTM +1 (non-binding)

File: ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java [hunk 2]
```diff
@@ -13967,7 +13968,8 @@ ASTNode analyzeCreateTable(
 
           isTransactional, isManaged, new String[]{qualifiedTabName.getDb(), qualifiedTabName.getTable()}, isDefaultTableTypeChanged);
       tblProps.put(hive_metastoreConstants.TABLE_IS_CTLT, "true");
-      isExt = isExternalTableChanged(tblProps, isTransactional, isExt, isDefaultTableTypeChanged);
+      isExt = isIcebergTable(tblProps) ||
+          isExternalTableChanged(tblProps, isTransactional, isExt, isDefaultTableTypeChanged);
```
Comment (deniskuzZ) [inline ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java:13972]: could we please move `isIcebergTable(tblProps)` check inside of isExternalTableChanged?

**Review by deniskuzZ — APPROVED** [review]
+1

**Pull request merged**